### ISSUE TITLE
[ENH] Implement Extended Isolation Forest

### ISSUE THREAD

**Issue by CodeLionX** [aeon-toolkit/aeon#2113]
### Describe the feature or idea you want to propose

The Extended Isolation Forest improves upon the Isolation Forest method.

### Describe your proposed solution

Implement directly in aeon:

- Paper: https://doi.org/10.1109/TKDE.2019.2947676
- Code: https://github.com/sahandha/eif
- Code (in TimeEval): https://github.com/TimeEval/TimeEval-algorithms/tree/main/eif

However, support semi-supervised as well as unsupervised usage similar to the [`PyODAdapter`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/anomaly_detection/_pyodadapter.py) or [`KMeansAD`](https://github.com/aeon-toolkit/aeon/blob/main/aeon/anomaly_detection/_kmeans.py)

**Comment by Akhil-Jasson** [comment]
Hi, I came across this issue and found it really interesting. I'm not entirely sure if I can solve it, but I'd love to give it a try. Would it be okay for me to work on this? Also, if there are any specific guidelines or things I should keep in mind while implementing this in Aeon, I'd appreciate the guidance.

**Comment by baraline** [comment]
Yes, you should use the base class of the module you are implementing the estimator to

**Comment by Akhil-Jasson** [comment]
![Image](https://github.com/user-attachments/assets/85cb8775-b46d-4fc3-8e76-d90ce2724121)
I'm encountering issues while importing the EIF model referenced above. When attempting to install the package, I receive a ModuleNotFoundError for numpy, even though I have already installed numpy in my environment. Could anyone suggest a fix? Alternatively, can I use  h2o.ai's EIF model as a viable alternative if this installation problem persists?

**Comment by Akhil-Jasson** [comment]
@MatthewMiddlehurst I'm trying to install the eif package mentioned in the issue above (https://github.com/sahandha/eif), but I'm facing the above difficulty.

**Comment by MatthewMiddlehurst** [comment]
This package has not been updated in 5 years. We generally prefer direct implementations rather than wrappers where possible. In this case I would reject code wrapping the package due to its unmaintained state. I believe the package was just linked as an example of an implementation.

**Comment by Akhil-Jasson** [comment]
@MatthewMiddlehurst  Since the eif package appears to be unmaintained, would using H2O.ai's implementation of EIF (https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/eif.html) be a viable alternative? Let me know if there are any preferred approaches for this use case

**Comment by MatthewMiddlehurst** [comment]
I have never used or heard of H2O.ai sorry. A wrapper of another package is better than nothing I guess, but the best option would be to have our own implementation. This algorithm seems to be widely available.

**Comment by Akhil-Jasson** [comment]
Please let me know if there's anything that needs to be modified or improved.
